playbook/app/pb_kits/playbook/pb_multi_level_select/_multi_level_select.tsx

```diff
@@ -78,6 +89,13 @@ const MultiLevelSelect = (props: MultiLevelSelectProps) => {
   useEffect(() => {
     if (returnAllSelected) {
       setReturnedArray(getCheckedItems(formattedData))
+    } else if (variant === "single") {
+      if (singleSelectedItem?.id?.length === 0) {
+        setSingleSelectedItem({ id: [], value: "", item: []})
+        setDefaultReturn(defaultReturn)
```

Comment:
not sure on this one @gavhuang , setting the state with it's own initial state doesn't seem like best practice. Why was this needed?

Reply:
Going to refactor this!